### ISSUE TITLE
Tolerance seems to have no effect (atleast on a circle)

### ISSUE THREAD

**Issue by rroohhh** [nical/lyon#646]
As exemplified by this example, the tolerance option doesn't seem to have any effect (atleast when tessellating a circle).

```rust
use lyon::math::{point, Point};
use lyon::tessellation::geometry_builder::simple_builder;
use lyon::tessellation::{FillOptions, FillTessellator, VertexBuffers};

fn main() {
    for tolerance in &[100.0, 0.000001] {
        let mut geometry: VertexBuffers<Point, u16> = VertexBuffers::new();
        let mut geometry_builder = simple_builder(&mut geometry);
        let mut tessellator = FillTessellator::new();

        tessellator.tessellate_circle(
            point(0.0, 0.0),
            0.5,
            &FillOptions::tolerance(*tolerance),
            &mut geometry_builder,
        );

        assert_eq!(geometry.vertices.len(), 8);
    }
}
```

**Comment by nical** [comment]
Thanks for filing this. Indeed the tolerance threshold got accidentally discarded in this methoid and the ellipse equivalent causing the two methods to always tessellate with the default 0.1 threshold.

**Comment by nical** [comment]
Fixed in lyon_tessellation 0.17.6.
